code/modules/shuttle/emergency.dm

```diff
@@ -253,31 +268,26 @@
 	// How did you even get on the shuttle before it go to the station?
 	if(!IS_DOCKED)
 		return FALSE
-	if(!..() || ENGINES_STARTED)
+	if(ENGINES_STARTED)
 		//SYSTEM ERROR: THE SHUTTLE WILL LA-SYSTEM ERROR: THE SHUTTLE WILL LA-SYSTEM ERROR: THE SHUTTLE WILL LAUNCH IN 10 SECONDS
 		to_chat(user, "<span class='warning'>The shuttle is already launching!</span>")
 		return FALSE
+	if((emag_last_used + emag_cooldown) >= world.time) //if emagging is on cooldown
+		say("Error - Catastrophic software error detected. Input is currently on timeout.")
+		return FALSE
 	return TRUE
 
 /obj/machinery/computer/emergency_shuttle/on_emag(mob/user)
 	..()
 	var/time = TIME_LEFT
-	message_admins("[ADMIN_LOOKUPFLW(user.client)] has emagged the emergency shuttle, [time] seconds before launch.")
-	log_game("[key_name(user)] has emagged the emergency shuttle in [COORD(src)] [time] seconds before launch.")
-
-	SSshuttle.emergency.movement_force = list("KNOCKDOWN" = 60, "THROW" = 20)//YOUR PUNY SEATBELTS can SAVE YOU NOW, MORTAL
-	var/datum/species/S = new
-	for(var/i in 1 to 10)
-		// the shuttle system doesn't know who these people are, but they
-		// must be important, surely
-		var/obj/item/card/id/ID = new(src)
-		var/datum/job/J = pick(SSjob.occupations)
-		ID.registered_name = S.random_name(pick(MALE, FEMALE))
-		ID.assignment = J.title
-
-		authorized += ID
-
-	process(SSMACHINES_DT)
+	if((obj_flags & EMAGGED))
```

Comment:
You need to adjust can_emag to allow this to happen, you shouldn't check conditions like this in on_emag. our emag code is a little different